build/tasks/build.js

```diff
@@ -3,125 +3,158 @@ var runSequence = require('run-sequence');
+    this.push(file);
+    return callback();
+  });
+}
+
+gulp.task('build-index', ['copy-resources'], function() {
+  var importsToAdd = paths.importsToAdd.slice();
 
-// concats submodules into one file, compiles d.ts file and copies them to the dist folders
-gulp.task('build-dts', function() {
-  var importsToAdd = []; // stores extracted imports
+  var src = gulp.src(paths.files);
 
-  return gulp.src(paths.tsSource)
-    //.pipe(tools.sortFiles()) can't sort with excluded files of the tsSource
-    .pipe(through2.obj(function(file, enc, callback) {  // extract all imports to importsToAdd
+  if (paths.sort) {
+    src = src.pipe(tools.sortFiles());
+  }
+
+  if (paths.ignore) {
+    paths.ignore.forEach(function(filename){
+      src = src.pipe(gulpIgnore.exclude(filename));
+    });
+  }
+
+  return src.pipe(through2.obj(function(file, enc, callback) {
       file.contents = new Buffer(tools.extractImports(file.contents.toString('utf8'), importsToAdd));
       this.push(file);
       return callback();
     }))
     .pipe(concat(jsName))
-    .pipe(to5(assign({}, compilerOptions.dts())));
-});
-
-gulp.task('copy-dts', function() {
-  var tdsPath = paths.packageName + '/' + paths.packageName + '.d.ts';
-  return gulp.src(tdsPath)
-      .pipe(rename(paths.packageName + '.d.ts'))
-      .pipe(gulp.dest(paths.output + 'es2015'))
-      .pipe(gulp.dest(paths.output + 'commonjs'))
-      .pipe(gulp.dest(paths.output + 'amd'))
-      .pipe(gulp.dest(paths.output + 'system'));
+    .pipe(insert.transform(function(contents) {
+      return tools.createImportBlock(importsToAdd) + contents;
+    }))
+    .pipe(gulp.dest(paths.output));
 });
 
-gulp.task('remove-dts-folder', function() {
-  var tdsFolder = paths.packageName;
-  return gulp.src([tdsFolder])
-    .pipe(vinylPaths(del));
+function gulpFileFromString(filename, string) {
+  var src = require('stream').Readable({ objectMode: true });
+  src._read = function() {
+    this.push(new gutil.File({ cwd: paths.appRoot, base: paths.output, path: filename, contents: new Buffer(string) }))
+    this.push(null)
+  }
+  return src;
+}
+
+function srcForBabel() {
+  return merge(
+    gulp.src(paths.output + '*.js'),
+    gulpFileFromString(paths.output + 'index.js', "export * from './" + paths.packageName + "';")
+  );
+}
+
+function srcForTypeScript() {
+  return gulp
+    .src(paths.output + paths.packageName + '.js')
+    .pipe(rename(function (path) {
+      if (path.extname == '.js') {
+        path.extname = '.ts';
+      }
+    }));
+}
+
+compileToModules.forEach(function(moduleType){
+  gulp.task('build-babel-' + moduleType, function () {
+    return srcForBabel()
+      .pipe(to5(assign({}, compilerOptions[moduleType]())))
+      .pipe(cleanGeneratedCode())
+      .pipe(gulp.dest(paths.output + moduleType));
+  });
+
+  if (moduleType === 'native-modules') return; // typescript doesn't support the combination of: es5 + native modules
```

Comment:
oh, i just copy the build from aurelia. but if you can help with the webpack and cli issues ... :smile: